lib/dialer/public.go

```diff
@@ -75,3 +78,22 @@ func SetTCPOptions(conn net.Conn) error {
 		return fmt.Errorf("unknown connection type %T", conn)
 	}
 }
+
+func SetTrafficClass(conn net.Conn, class int) error {
+	switch conn := conn.(type) {
+	case *net.TCPConn:
+		e1 := ipv4.NewConn(conn).SetTOS(class)
+		e2 := ipv6.NewConn(conn).SetTrafficClass(class)
```

Comment:
I guess one of these will always fail as the connection is either one or the other?